app/models/report.rb

```diff
@@ -211,6 +219,18 @@ def data_multiple_departments
     }
   end
 
+  def data_student_contributions
+    row_data = {}
+    terms = Thesis.all.pluck(:grad_date).uniq.sort
+    terms.each do |term|
+      row_data[term] = Thesis.where('grad_date = ?', term).map(&:student_contributed?).count(true)
```

Comment:
Just curious, what does passing the boolean to `count` do here? Is it explicitly counting the truthy values resulting from `map(&:student_contributed?`

Reply:
Yeah, that's a way of counting only the records for which `student_contributed?` is true - without that boolean you get back the list of thesis records for each term. There may be a more efficient way of generating this count, but I'm not sure what it is.

I think there's a ticket in the backlog for dealing with performance of the reporting area, so I'm trying not to worry unduly about performance until that comes up.

Reply:
Quick addendum - there are three ways that I've handled querying for these reports thus far, but the one in this method is the best one I can come up with at the moment:

1. grab list of terms, iterate over each and select qualifying records (that's the method used here)
2. grab all qualifying records, segmented by term, and then trust in `pad_terms()` to fill in zeros as appropriate. This is the approach taken in `data_thesis_records` below, but I'm unable to make it work for this ticket (yet?)
3. write the underlying SQL ourselves. I try pretty hard to avoid this approach, because it feels fragile and not taking advantage of Rails, but in certain cases it should allow one SQL query to return precisely the records we need. This is seen in the method above this one, `data_multiple_departments`

I'd love to see more methods use option 2, as the code is cleanest in those cases, but I'm not sure that's always the most performant option. I kind of suspect that option 3 is the best for performance, with option 2 as a middle ground and option 1 as the "least performant but most capable" choice.